### DISCUSSION TITLE
Superscript

### DISCUSSION THREAD

**Discussion by wartburggraaf** [nl-design-system/backlog#336] [Component suggestions]
<img width="480" alt="visuele weergave van een Superscript component" src="https://github.com/user-attachments/assets/06fa0505-d2a8-4007-9fbb-68f45a0b4049">

We willen graag meer leren over de Superscript component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook, Github of screenshots. Met deze informatie kunnen we bewijzen dat de component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1725519660424549)

---

## Naam
Superscript

## Doel

Toont tekst met een kleiner lettertype met een verhoogde letterlijn.

## Gerelateerde componenten

[Subscript](https://github.com/orgs/nl-design-system/discussions/334)

## Onderzoek
[Subscript and superscript - Wikipedia](https://en.wikipedia.org/wiki/Subscript_and_superscript)
Wikipedia pagina waarbij ook vele voorbeelden van gebruik worden beschreven.

[`<sup>`: The Superscript element - MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/sup)
Documentatie van Mozilla Developers Network over het `<sup>` HTML element.

[Problems solved by OpenType | Roel Nieskens | CSS Day 2024 - Youtube](https://www.youtube.com/watch?v=TreBK-EyACQ&t=1520s)
Riel Nieskens deelt een presentatie bordevol tips en toont onder andere hoe je super- en subscript in CSS kan gebruiken.

[Brief Note on Super- and Subscript Text - Adrian Roselli](https://adrianroselli.com/2022/09/brief-note-on-super-and-subscript-text.html)
Adrian Roselli heeft onderzoek gedaan naar support voor super- en subscript door assistieve technologie.

**Comment by jeffreylauwers** [comment]
## Utrecht

- [Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-superscript--docs)
- [GitHub](https://github.com/nl-design-system/utrecht/tree/main/components/superscript)

---

### Screenshots / Varianten

**Storybook**

Default
![Scherm­afbeelding 2024-09-27 om 13 17 02](https://github.com/user-attachments/assets/1ed19d00-a26b-41d1-83a1-55177a4af370)

Superscript inside multiple lines of text
![Scherm­afbeelding 2024-09-27 om 13 17 08](https://github.com/user-attachments/assets/7bad573c-fcc3-45be-97e5-44bf6855fbbb)

Superscript with Open Type rendering, inside multiple lines of text
![Scherm­afbeelding 2024-09-27 om 13 17 17](https://github.com/user-attachments/assets/2e08406e-7a53-42d3-b5b8-e5f24da37fd3)

---

### Notities

Documentatie aanwezig (Storybook)

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Superscript op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=superscript)

**Comment by rianrietveld** [comment]
Klein onderzoekje naar screenreader feedback op` <sup>`.
Zie ook vergelijkbare tekst bij de [sub-discussie](https://github.com/orgs/nl-design-system/discussions/334).

TL;DR: screenreaders geven geen info over sub of sub.

Alleen VoiceOver probeert er wat van het maken als deze de structuur herkent.
Dus `12 m<sup>2</sup>` wordt als twaalf vierkante meter uitgesproken. Maar `3<sup>2</sup>` wordt voorgelezen als 32.
NVDA wacht even voor het uitspreken van de sub en sup maar noemt alleen het nummer op.
JAWS negeert het volkomen en maakt het onderdeel van de tekst.
Dit valt dus in dezelfde categorie als `<b>` em `<i>`.
Alleen visuele informatie die niet wordt doorgegeven aan screen readers.
Ik zou deze component afraden voor content.
Zie voor testcode:
https://codepen.io/rianrietveld/pen/xxvPEar

Discussie met @Robbert
unicode karakters als m². VoiceOver leest 12² voor als “12 digit 2”.  Zonder de sup-info.
We moeten de veelvoorkomende use cases uitsplitsen, en een advies moeten geven per onderdeel zoals:

- vierkante en cubbice meters in m2 en m3, etc
- wiskundige formules
- voetnoot verwijzingen
[á la Julesprudentie](https://www.200ok.nl/tips/speciale-tekens-en-screenreaders/)
H2O lijkt me bijvoorbeeld harmless, voor formules moeten we misschien figures + figcaption + mathml adviseren

Aan content managers adviseren de info uit te schrijven: twaalf vierkante meter, ipv `12m<sup>2</sup>`.

Hidde:
Er zijn wel mappings in HTML-AAM ([sub](https://www.w3.org/TR/html-aam-1.0/#el-sub) en [sup](https://www.w3.org/TR/html-aam-1.0/#el-sup)), dus dit zijn screenreader bugs. Adrian heeft een [issue gefiled bij JAWS](https://github.com/FreedomScientific/standards-support/issues/668), lijkt weinig te gebeuren.

Nadeel van geen component is denk ik dat als in de toekomst support beter wordt we geen updates aan een component kunnen doen? Maar niet echt een nadeel als het alternatief 'uitschrijven' is en dat ook gewoon goed genoeg werkt

**Comment by jeffreylauwers** [comment]
Anton Olivier noemde tijdens de Design Open Dag ook Copyright en Trademark als eventuele toepassingen.

https://www.dictionary.com/e/trademark-copyright-registered-symbols/
